web/src/site-admin/SiteAdminActivation.tsx

```diff
@@ -0,0 +1,259 @@
+    private enableSignOn = () => {
+        window.open('https://docs.sourcegraph.com/admin/auth', '_blank')
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="site-admin-checklist-container">
+                {this.props.checklistInfo ? (
+                    <ul className="site-admin-checklist">
+                        <li>
+                            <ChecklistItem
+                                title="Connect your code host"
+                                done={this.props.checklistInfo.connectedCodeHost}
+                                action={this.addCodeHosts}
+                                detail="Configure Sourcegraph to talk to your code host and fetch a list of your repositories."
+                            />
+                        </li>
+                        <li>
+                            <ChecklistItem
+                                title="Enable repositories"
+                                done={this.props.checklistInfo.enabledRepository}
+                                action={this.enableRepos}
+                                detail="Select which repositories Sourcegraph should pull and index from your code host(s)."
+                            />
+                        </li>
+                        <li>
+                            <ChecklistItem
+                                title="Enable an extension"
+                                done={this.props.checklistInfo.enabledExtension}
+                                action={this.enableExtension}
+                                detail="Enable a Sourcegraph extension to add jump-to-def, find-refs, or helpful annotations from 3rd party dev tools."
+                            />
+                        </li>
+                        <li>
+                            <ChecklistItem
+                                title="Configure sign-up"
+                                done={this.props.checklistInfo.enabledSignOn}
+                                action={this.enableSignOn}
+                                detail="Configure a single-sign on (SSO) provider or enable open sign-up so you can share Sourcegraph with others on your team."
+                            />
+                        </li>
+                    </ul>
+                ) : (
+                    <div>Loading...</div>
+                )}
+            </div>
+        )
+    }
+}
+
+interface ChecklistItemProps {
+    title: string
+    action: () => void
+    done?: boolean
+    detail?: string
+}
+
+interface ChecklistItemState {
+    showDetail?: boolean
+}
+
+/**
+ * ChecklistItem is a single item in the site admin activation checklist.
+ */
+class ChecklistItem extends React.PureComponent<ChecklistItemProps, ChecklistItemState> {
+    public state: ChecklistItemState = {}
+
+    private toggleDetail = (e: React.MouseEvent<HTMLSpanElement>) => {
+        e.stopPropagation()
+        this.setState({ showDetail: !this.state.showDetail })
```

Comment:
Never reference `this.state` in `setState()` input because it is async. Use the callback form
```suggestion
        this.setState(state => ({ showDetail: !state.showDetail }))
```